sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -193,10 +193,12 @@ private[sql] object Dataset {
  */
 @Stable
 class Dataset[T] private[sql](
-    @DeveloperApi @Unstable @transient val queryExecution: QueryExecution,
+    @DeveloperApi @Unstable @transient private[sql] var queryExecution: QueryExecution,
     @DeveloperApi @Unstable @transient val encoder: Encoder[T])
   extends Serializable {
 
+  def queryExecution(): QueryExecution = queryExecution
```

Comment:
is this binary compatibility?

Reply:
I'm not familiar with this, can anyone help take a look?